lib/index.ts

```diff
@@ -130,32 +139,50 @@ class LogService {
     }
   }
 
+  private applyFormat(logger: LoggerPlugin, formatOptions: FormatOptions) {
+    const { transform } = combine(
+      label({ label: process.pid.toString() }),
+      timestamp(),
+      formatOptions.prettify ? prettyPrint({ depth: 4 }) : this.simplifyPrint()
+    );
+
+    for (const level of Object.values(LogLevel)) {
+      const oldLoggerMethod = logger[level];
+      logger[level] = (msg: any) => {
+        const transformedMsg = transform({ level: level, message: msg });
+        const symbolKey = Object.getOwnPropertySymbols(transformedMsg)[0];
+        oldLoggerMethod((transformedMsg as any)[symbolKey]);
+      };
+    }
+  }
+
   addLogger(
-    loggerScopeFunction: (config: InitialConfig) => LoggerPlugin,
-    pluginConfig: PluginConfig) {
-
-    const loggerPlugin = loggerScopeFunction.call(this, this.globalConfig);
-    const loggerPluginKeys = Object.keys(loggerPlugin ?? {});
-    const logLevel = ['error', 'warn', 'info', 'debug'];
-    const missingMethods: String[] = [];
-    const isValidObject = logLevel.filter((method) => {
-      if (loggerPluginKeys.indexOf(method) === -1) missingMethods.push(method);
-      return true;
-    }).every((method) => loggerPluginKeys.indexOf(method) !== -1)
-
-    if (!isValidObject)
+    loggerPluginMaker: (config: InitialConfig) => LoggerPlugin,
+    pluginConfig: PluginConfig = { level: LogLevel.debug, silent: false }
+  ) {
+    const loggerPlugin = loggerPluginMaker.call(this, this.globalConfig);
+    const loggerPluginMethods = Object.keys(loggerPlugin ?? {});
+    const requiredMethods = Object.values(LogLevel);
+
+    const missingMethods = requiredMethods.filter(
+      (method) => !loggerPluginMethods.includes(method)
+    );
+
+    if (missingMethods.length)
       throw new Error(
-        `Invalid Logger, missing one or more of required methods: [${missingMethods.join(', ')}]`
+        `Invalid Logger, missing these required methods: [${missingMethods.join(', ')}]`
```

Comment:
Here we can't say "these" because if it's only one method it won't make a lot of sense... I believe that "one or more required methods" is good approach